Address eellison's CR comments on AOTAutograd

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85370

For some reason, this change didn't actually make it to master.

Signed-off-by: Edward Z. Yang <ezyangfb.com>

ezyang committed:
Address eellison's CR comments on AOTAutograd

For some reason, this change didn't actually make it to master.

Signed-off-by: Edward Z. Yang <ezyangfb.com>

[ghstack-poisoned]

vadimkantorov commented on functorch/_src/aot_autograd.py:
```diff
@@ -312,25 +312,31 @@ def aot_dispatch_autograd(flat_fn, flat_args: List[Tensor], aot_config: AOTConfi
-            if i in drop_args:
-                continue
-            r.append(t)
+        for t, keep in zip(args, keep_arg_mask):
```
is simpler as a list comprehension? `return [t for t, keep in zip(args, keep_arg_mask) if keep]`

same for a function below?

ezyang replied:
sure

ezyang committed:
Update on "Address eellison's CR comments on AOTAutograd"


For some reason, this change didn't actually make it to master.

Signed-off-by: Edward Z. Yang <ezyangfb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Address eellison's CR comments on AOTAutograd"


For some reason, this change didn't actually make it to master.

Signed-off-by: Edward Z. Yang <ezyangfb.com>

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge

ezyang committed:
Update on "Address eellison's CR comments on AOTAutograd"


For some reason, this change didn't actually make it to master.

Signed-off-by: Edward Z. Yang <ezyangfb.com>

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge -f "ci flakiness"
